src/student/student.service.ts

```diff
@@ -134,200 +135,164 @@ export class StudentService {
+            .select('projects.url', 'portfolioUrl')
+            .from(ProjectEntity, 'projects')
+            .where('projects.user_id = :userId', { userId })
+            .andWhere('projects.type="1"');
+        }, 'portfolioUrl')
+        .getRawOne();
+      console.log(student);
+      return {
+        studentDetails: student,
+      };
+    } catch {
+      throw new InternalServerErrorException();
+    }
   }
 
-  async findOne(id: string): Promise<StudentProfileResponse> {
-    const userId = await this.getUserByStudentId(id);
-
-    const student = await this.studentRepository
-      .createQueryBuilder('student')
-      .leftJoin('student.user', 'user')
-      //.leftJoin('student.projects','projects')
-      .select([
-        'user.email',
-        'student.phoneNumber',
-        'student.firstName',
-        'student.lastName',
-        'student.githubName',
-        'student.bio',
-        'student.expectedWorkType',
-        'student.targetWorkCity',
-        'student.expectedContractType',
-        'student.expectedSalary',
-        'student.canTakeApprenticeship',
-        'student.monthsOfCommercialExperience',
-        'student.education',
-        'student.workExperience',
-        'student.courses',
-      ])
-      .addSelect((subQuery) => {
-        return subQuery
-          .select('projects.url', 'bonusProjectUrl')
-          .from(ProjectEntity, 'projects')
-          .where('projects.user_id = :userId', { userId })
-          .andWhere('projects.type="0"');
-      }, 'bonusProjectUrl')
-      .addSelect((subQuery) => {
-        return subQuery
-          .select('projects.url', 'portfolioUrl')
-          .from(ProjectEntity, 'projects')
-          .where('projects.user_id = :userId', { userId })
-          .andWhere('projects.type="1"');
-      }, 'portfolioUrl')
-      .where('student.id = :id', { id })
-      .getRawOne();
-
-    if (!student) {
-      throw new NotFoundException('Kursant z podanym id nie istnieje');
+    async findStudentByUserId(id: string): Promise<StudentEntity> {
+        try {
+            return await this.studentRepository.findOne({where: {userId: id}})
+        } catch {
+            throw new InternalServerErrorException();
+        }
     }
-    console.log(student);
-    return {
-      studentDetails: student,
-    };
-  }
 
-  async updateOne(
-    studentId: string,
-    studentProfileDetails: UpdateStudentDetailsDto,
-  ): Promise<UpdatedStudentResponse> {
-    const student = await this.studentRepository.findOne({
-      where: { id: studentId },
-    });
+    async updateOne(student, studentProfileDetails: UpdateStudentDetailsDto): Promise<UpdatedStudentResponse> {
+        try {
+            const dataUpdatedAt = () => 'CURRENT_TIMESTAMP';
 
-    const dataUpdatedAt = () => 'CURRENT_TIMESTAMP';
+            const {
+                email,
+                bonusProjectUrl,
+                portfolioUrl,
+                githubName,
+                updatedAt,
```

Comment:
Trzeba by zmienić dto aby z frontu nie trzeba było podawać czasu (updatedAt). Czas update'u tabeli powinien podać sam backend, więc nie musimy go przyjmować z frontu.

Reply:
ja w sumie przesyłałam to zawsze z BE, tylko do DTO nie potrzebnie trafiło - po zmianach powinno być ok